htlcswitch/link.go

```diff
@@ -2137,6 +2159,29 @@ func (l *channelLink) handleUpstreamMsg(msg lnwire.Message) {
 			"ChannelPoint(%v): received error from peer: %v",
 			l.channel.ChannelPoint(), msg.Error(),
 		)
+
+	case *lnwire.Shutdown:
+		// We've received a Shutdown message from the remote peer.
+		// We'll set the shutdownReceived bool and cancel back any new
+		// HTLC's received after this point instead of forwarding them.
+		if l.shutdownReceived {
+			// Fail the link if a duplicate shutdown is received.
+			// This is also checked in peer.Brontide, but done here
+			// to properly stop the link.
+			l.fail(
+				LinkFailureError{code: ErrRemoteError},
+				"ChannelPoint(%v): received dupe shutdown "+
+					"from peer", l.channel.ChannelPoint(),
+			)
+
+			return
+		}
+
+		// Otherwise, set the shutdownReceived bool.
```

Comment:
This "don't allow a second shutdown message" logic has already been implemented in `chancloser` and I think we should avoid duplicating it.

Reply:
This is done to stop the link rather than have the `chancloser` stop the link